### PR TITLE
#314439 -- #791 Fixes button shifting on hover in HC mode

### PR DESCRIPTION
**Pull request by aleksandrjPersonal**
#### Pull request checklist

- [X ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss [hunk 1]
```diff
@@ -136,6 +136,8 @@ $SearchBox-sidePadding: 8px; // padding in input on left and right sides without
     &.is-expanded {
       @media screen and (-ms-high-contrast: active) {
         @include highContrastBorder($ms-color-white);
+        margin-left: -1px;
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss]: What issue is this fixing? I'm guessing the problem is in the `highContrastBorder()` mixin and that we shouldn't be using negative margins here.

**Comment by aleksandrjPersonal** [comment]
Based on suggestions/work with Mike Weaton, we are using a pseudo element to draw a box over the CommandBarItem. This allow us to remove rules for adding borders and adjusting heights

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss [hunk 2]
```diff
@@ -37,8 +37,23 @@ $SearchBox-iconSize: 16px;
 $SearchBox-sidePadding: 8px; // padding in input on left and right sides without icons
 
 @mixin highContrastBorder($color) {
-    @include border-left(1px, solid, $color);
-    @include border-right(1px, solid, $color);
+  position: relative;
+
+  &::after {
+    // Wrap the parent element with $padding pixels.
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss]: We forgot to delete this comment.

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss [hunk 3]
```diff
@@ -131,12 +134,11 @@ $SearchBox-sidePadding: 8px; // padding in input on left and right sides without
   .ms-CommandBarItem-link {
     @include CommandBarItem-text;
     cursor: pointer;
-
     &:hover,
     &.is-expanded {
       @media screen and (-ms-high-contrast: active) {
-        @include highContrastBorder($ms-color-white);
-      }
+         @include highContrastBorder($ms-color-white);
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss]: Can you double-check the indentation here (2 spaces)? I'm not sure why GitHub is showing changes on these lines.

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss (hunk 2 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss]: ahh ok

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss (hunk 3 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.scss]: yes

**Pull request closed without merging**